### PR TITLE
Date package: Keep updated localeData in momentLib

### PR DESCRIPTION
**Pull request by psealock**
## Description
This PR modifies `wp.date.setSettings` such that `momentLib` retains updated `i18n` data when a page initially sets settings.

Currently the code correctly updates `momentLib` with locale data from settings but simply switches back to the `currentLocale` which will always be the default "en". 

I'm not entirely sure why this code was there in the first place. Git blame didn't reveal much so I can't see the history there. 

## Testing instructions

1. Switch languages to something other than English
2. Open the console to any page with Gutenberg
3. Copy/paste the following code into the console
```js
var dateFormat = wp.date.getSettings().formats.date;
var today = wp.date.moment();
wp.date.date( dateFormat, today );
```
If using French, the output should be `"26 juin 2018"`. 
The master branch will output the default language "en" `"26 June 2018"`.

Alternatively, you can paste `moment.localeData()` and view `momentLib`'s data. It should reflect the language setting in site settings.

## Types of changes
This changes introduced allow the `moment` global to reflect language settings in site settings

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 

cc @gziolo

### CODE REVIEW

**Comment by psealock** [comment]
I've also added `start_of_week` and placed it into the `momentLib` locale instance so that use of Moment surrounding weeks will be correctly calculated

File: packages/date/src/index.js [hunk 1]
```diff
@@ -71,8 +71,10 @@ export function setSettings( dateSettings ) {
 			y: 'a year',
 			yy: '%d years',
 		},
+		week: {
+			dow: dateSettings.l10n.start_of_week,
+		},
 	} );
-	momentLib.locale( currentLocale );
```
Comment (youknowriad) [inline packages/date/src/index.js]: I wonder if we should keep this and apply it in `wp.data.moment` instead. I think the idea of this is that the module is trying to avoid messing with any external global configuration applied to moment. 

Also, it might be good to see how we can get rid of moment entirely from the external API of this module (but this is out of scope of this PR)

Comment (psealock) [reply packages/date/src/index.js]: Thanks @youknowriad, I understand now. So an extension can simply switch or update the locale information on its own.

```js
// Switch to the previously loaded locale
moment.locale( wp.date.getSettings().l10n.locale );

// Or even update the data if need be
moment.updateLocale(  wp.date.getSettings().l10n.locale, { ... }  );
```
This satisfies the needs of our app. I'll close this PR, thanks for you help.

>it might be good to see how we can get rid of moment entirely from the external API

I can agree here. From the perspective of someone consuming the API, its causing more confusion than help. In fact, even switching to the previously loaded locale causes issues, e.g. this breaks usage of moment's format api, `moment().format( 'LL' )` because these formats in `dateSettings`are PHP based.

```js
longDateFormat: {
    LT: dateSettings.formats.time,
    LTS: null,
    L: null,
    LL: dateSettings.formats.date,
    LLL: dateSettings.formats.datetime,
    LLLL: null,
}
```
